lib/github_changelog_generator/generator/entry.rb

```diff
@@ -0,0 +1,232 @@
+
+      begin
+        sections_json = JSON.parse(sections_desc)
+      rescue JSON::ParserError => e
+        raise "There was a problem parsing your JSON string for secions: #{e}"
+      end
+
+      sections_arr = []
+
+      sections_json.each do |name, v|
+        sections_arr << Section.new(name: name.to_s, prefix: v["prefix"], labels: v["labels"], options: @options)
+      end
+
+      sections_arr
+    end
+
+    # Creates a hash map of labels => section objects
+    #
+    # @return [Hash] map of labels => section objects
+    def label_map
+      @sections.each_with_object({}) do |section_obj, memo|
+        section_obj.labels.each do |label|
+          memo[label] = section_obj.name
+        end
+      end
+    end
+
+    # Creates a hash map of 'section name' => section object
+    #
+    # @return [Hash] map of 'section name' => section object
+    def section_map
+      @sections.each_with_object({}) do |section, memo|
+        memo[section.name] = section
+      end
+    end
+
+    # It generates header text for an entry with specific parameters.
+    #
+    # @param [String] newer_tag_name - name of newer tag
+    # @param [String] newer_tag_link - used for links. Could be same as #newer_tag_name or some specific value, like HEAD
+    # @param [Time] newer_tag_time - time, when newer tag created
+    # @param [String] older_tag_name - tag name, used for links.
+    # @param [String] project_url - url for current project.
+    # @return [String] - Header text for a changelog entry.
+    def generate_header(newer_tag_name, newer_tag_link, newer_tag_time, older_tag_name, project_url)
+      header = ""
+
+      # Generate date string:
+      time_string = newer_tag_time.strftime(@options[:date_format])
+
+      # Generate tag name and link
+      release_url = if @options[:release_url]
+                      format(@options[:release_url], newer_tag_link)
+                    else
+                      "#{project_url}/tree/#{newer_tag_link}"
+                    end
+      header += if newer_tag_name.equal?(@options[:unreleased_label])
+                  "## [#{newer_tag_name}](#{release_url})\n\n"
+                else
+                  "## [#{newer_tag_name}](#{release_url}) (#{time_string})\n\n"
+                end
+
+      if @options[:compare_link] && older_tag_name
+        # Generate compare link
+        header += "[Full Changelog](#{project_url}/compare/#{older_tag_name}...#{newer_tag_link})\n\n"
+      end
+
+      header
+    end
+
+    # Generates complete body text for a tag (without a header)
+    #
+    # @param [Array] pull_requests
+    # @param [Array] issues
+    # @returns [String] ready-to-go tag body
+    def generate_body(pull_requests, issues)
+      body = ""
+      body += main_sections_to_log(pull_requests, issues)
+      body += merged_section_to_log(pull_requests) if (@options[:pulls] && @options[:add_pr_wo_labels]) || (@options.configure_sections? && @options[:include_merged])
```

Comment:
It looks like the logic here is not correct. `:include_merged` defaults to true, but seems like it should default to false. Additionally, the `:configure_section` option does not override the `:add_pr_wo_labels` option as it appears to be meant to do.

The UX here needs a little thinking through.